### PR TITLE
feat: add hardware column in tests table

### LINKED ISSUE #408: Add environment compatibles in tests table
Add only the first hardware to the table and put the rest in the hover

### PR DESCRIPTION
**Pull request by anajalvarenga**
add hardware column in tests table

- Closes #408

### CODE REVIEW

File: dashboard/src/pages/TreeDetails/Tabs/Tests/IndividualTestsTable.tsx [hunk 1]
```diff
@@ -79,12 +80,48 @@ const columns: ColumnDef<TIndividualTest>[] = [
...
 ];
 
+const TooltipHardware = ({
+  hardwares,
+}: {
+  hardwares: string[] | undefined;
+}): JSX.Element => {
+  return (
+    <Tooltip>
+      <TooltipTrigger>{hardwares ? hardwares[0] : '-'}</TooltipTrigger>
```
Comment (lfjnascimento) [inline dashboard/src/pages/TreeDetails/Tabs/Tests/IndividualTestsTable.tsx:110]: you can use `sanitizeTableValue(hardwares[0], false)` just to have the empty value `'-'` being set in one place

File: dashboard/src/pages/TreeDetails/Tabs/Tests/IndividualTestsTable.tsx [hunk 2]
```diff
@@ -79,12 +80,48 @@ const columns: ColumnDef<TIndividualTest>[] = [
...
+      <TooltipContent>
+        {hardwares
+          ? hardwares.map(hardware => (
```
Comment (lfjnascimento) [inline dashboard/src/pages/TreeDetails/Tabs/Tests/IndividualTestsTable.tsx:113]: we can memoize this array

**Review by lfjnascimento — APPROVED** [review]
Nice! Some comments, but I'm pre-approving

File: backend/kernelCI_app/views/treeDetailsSlowView.py [hunk 3]
```diff
@@ -155,14 +155,17 @@ def __getCurrentRowData(self, currentRow):
...
+            tempColumnDict["tests_environment_compatible"]
+        ] is not None else None
 
         historyItem = {
             "id": testId,
             "status": testStatus,
             "path": path,
             "duration": testDuration,
             "startTime": currentRow[tempColumnDict["tests_start_time"]],
+            "hardware": currentRow[tempColumnDict["tests_environment_compatible"]]
```
Comment (Francisco2002) [inline backend/kernelCI_app/views/treeDetailsSlowView.py:168]: ditto about reuse variable (`testEnvironmentCompatible`)

File: backend/kernelCI_app/views/treeDetailsSlowView.py [hunk 4]
```diff
@@ -155,14 +155,17 @@ def __getCurrentRowData(self, currentRow):
         testError = extract_error_message(currentRow[tempColumnDict["tests_misc"]])
         testEnvironmentCompatible = currentRow[
             tempColumnDict["tests_environment_compatible"]
-        ]
+        ][0] if currentRow[
```
[the remark this answers is not in the document]
Comment (anajalvarenga) [reply backend/kernelCI_app/views/treeDetailsSlowView.py:158]: `currentRow[tempColumnDict["tests_environment_compatible"]]` is an string array or None and here I only want one if it is an array

File: backend/kernelCI_app/views/treeDetailsSlowView.py [hunk 5]
```diff
@@ -155,14 +155,17 @@ def __getCurrentRowData(self, currentRow):
...
+            tempColumnDict["tests_environment_compatible"]
+        ] is not None else None
 
         historyItem = {
             "id": testId,
             "status": testStatus,
             "path": path,
             "duration": testDuration,
             "startTime": currentRow[tempColumnDict["tests_start_time"]],
```
[the remark this answers is not in the document]
Comment (anajalvarenga) [reply backend/kernelCI_app/views/treeDetailsSlowView.py:167]: In this issue I did not work with start time

File: backend/kernelCI_app/views/treeDetailsSlowView.py (hunk 3 above)
Comment (anajalvarenga) [reply backend/kernelCI_app/views/treeDetailsSlowView.py:168]: testEnvironmentCompatible gets only the first element and here I want them all

**Review by MarceloRobert — APPROVED** [review]
Looks good, that column will be useful to find where the hardwares are coming from

**Pull request merged**